pkg/crc/services/dns/dns_darwin.go

```diff
@@ -42,11 +41,12 @@ func runPostStartForOS(serviceConfig services.ServicePostStartConfig, result *se
 		return *result, err
 	}
 	if needRestart {
-		// Restart the Network on mac
+		// Restart the Network but in case of error log it to error info.
+		// If we make it blocking call then in offline use case for mac is
+		// always going to be broken.
 		logging.InfoF("Restarting the host network")
 		if err := restartNetwork(); err != nil {
-			result.Success = false
-			return *result, err
+			logging.ErrorF("Restarting the host network failed: %v", err)
```

Comment:
:+1:

Reply:
Alternatively, you could move the `waitForNetwork()` call here and ignore its errors, or return a different error type from `restartNetwork` depending on what failed.
Given the need to ignore the error, I'm under the impression that "ping 8.8.8.8" is not the check we want for checking if network restart is done or not (and I'd rather we don't let google know every time we start a crc VM on mac).

Reply:
> Alternatively, you could move the waitForNetwork() call here and ignore its errors, or return a different error type from restartNetwork depending on what failed.

@cfergeau you mean something like this?
```golang
if err := restartNetwork(); err != nil {
	logging.ErrorF("Restarting the host network failed: %v", err) => don't ignore this
}
if err := waitForNetwork(); err != nil {
         logging.ErrorF(err) => ignore it here ?
}
```

Reply:
> Given the need to ignore the error, I'm under the impression that "ping 8.8.8.8" is not the check we want for checking if network restart is done or not (and I'd rather we don't let google know every time we start a crc VM on mac).

@cfergeau what do you suggest a reliable nameserver we can use instead using google one?

Reply:
> > Alternatively, you could move the waitForNetwork() call here and ignore its errors, or return a different error type from restartNetwork depending on what failed.
> 
> @cfergeau you mean something like this?
> 
> ```go
> if err := restartNetwork(); err != nil {
> 	logging.ErrorF("Restarting the host network failed: %v", err) => don't ignore this
> }
> if err := waitForNetwork(); err != nil {
>          logging.ErrorF(err) => ignore it here ?
>```

yes :)

Reply:
> > Given the need to ignore the error, I'm under the impression that "ping 8.8.8.8" is not the check we want for checking if network restart is done or not (and I'd rather we don't let google know every time we start a crc VM on mac).
> 
> @cfergeau what do you suggest a reliable nameserver we can use instead using google one?

As I understand it, the problem we are trying to solve is that the network restart is asynchronous, and we need a way to know when the network is fully restarted and is up. We decided to do that by pinging a well known IP (to avoid potential DNS resolving issues), so we picked 8.8.8.8. for that. 4.2.2.2 might be slightly better (but at this point that's really nitpicking).

"ping to external IP addresses" might even be failing on setups where network connectivity is fine (proxy, machine without internet connection, ...). We could try to build something around `networksetup -getinfo` instead, but there will probably be cornercases where it will report failure when it should not. I dunno if native macos API give a way to handle this. I guess a better long-term solution will be to avoid this network restart so that we don't have to wait for it to complete ;)

Reply:
@cfergeau according to https://www.tummy.com/articles/famous-dns-server/ article (published in 2010) I shouldn't use `4.2.2.2` unless I am a customer of Level-3 this is not deliberately setup as a public dns server like google one.

Reply:
You should not use it as your day to day DNS server to resolve DNS queries unless you are a level-3 customer. The way I read this article is that you can use it to check for network connectivity.

Reply:
>Unless you are a Level-3 customer, absolutely not. Google now has established public DNS servers at 8.8.8.8 which you should use. [...]  It's also a compelling choice if you just need to send out a ping to see if you can reach the Internet, if you don't have IPs you already know like I do. [...] However, some people set up 4.2.2.[1-6] as their standard DNS servers. Don't do this unless you are connected to Level 3.